vassal-app/src/main/java/VASSAL/build/module/map/deck/SortParameterArrayConfigurer.java

```diff
@@ -0,0 +1,296 @@
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, copies are available
+ * at http://www.opensource.org.
+ */
+package VASSAL.build.module.map.deck;
+
+import VASSAL.configure.AbstractConfigurableListEntry;
+import VASSAL.configure.ConfigurableList;
+import VASSAL.configure.ConfigurableListController;
+import VASSAL.configure.ConfigurableListEntry;
+import VASSAL.configure.Configurer;
+import VASSAL.configure.ConfigurerLayout;
+import VASSAL.counters.TraitLayout;
+import VASSAL.tools.SequenceEncoder;
+
+import java.awt.Component;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.swing.BorderFactory;
+import javax.swing.JComponent;
+import javax.swing.JPanel;
+
+import net.miginfocom.swing.MigLayout;
+
+public class SortParameterArrayConfigurer extends Configurer implements ConfigurableList {
+
+  private JPanel controls;
+  private JPanel panel;
+  private int selectedEntryIndex = -1;
+  private ConfigurableListController controller;
+  private final List<SortParameterEntry> entries = new ArrayList<>();
+
+  public SortParameterArrayConfigurer(String key, String name) {
+    this(key, name, null);
+  }
+
+  public SortParameterArrayConfigurer(String key, String name, Object val) {
+    super(key, name, val);
+  }
+
+  @Override
+  public void moveEntryUp() {
+    if (getSelectedEntryIndex() < 0) {
+      return;
+    }
+
+    final int pos = getSelectedEntryIndex();
+    final List<SortParameter> params = getValue() == null ? new ArrayList<>() : getSortParameterListValue();
+    final SortParameter moving = params.remove(pos);
+    params.add(pos - 1, moving);
+    setValue(params);
+
+    setSelectedEntryIndex(pos - 1);
+
+    rebuildControls();
+  }
+
+  @Override
+  public void moveEntryDown() {
+    if (getSelectedEntryIndex() < 0) {
+      return;
+    }
+
+    final int pos = getSelectedEntryIndex();
+    final List<SortParameter> params = getValue() == null ? new ArrayList<>() : getSortParameterListValue();
+    final SortParameter moving = params.remove(pos);
+    params.add(pos + 1, moving);
+    setValue(params);
+
+    setSelectedEntryIndex(pos + 1);
+
+    rebuildControls();
+  }
+
+  @Override
+  public void addEntry() {
+    final int pos = getSelectedEntryIndex();
+    final List<SortParameter> params = getValue() == null ? new ArrayList<>() : getSortParameterListValue();
+
+    int newEntry;
+    // Insert the new entry into the list at the appropriate place
+    if (entries.isEmpty() || getSelectedEntryIndex() < 0) {
+      params.add(new SortParameter());
+      setValue(params);
+      newEntry = getSortParameterListValue().size() - 1;
+      setSelectedEntryIndex(newEntry);
+    }
+    else {
+      newEntry = pos + 1;
+      params.add(pos, new SortParameter());
+      setValue(params);
+      setSelectedEntryIndex(newEntry);
```

Comment:
You could put `setSelectedEntryIndex(newEntry);` outside the case structure here, since it's last in both cases. Maybe also `setValue(params)`?

Reply:
setSelectedEntryIndex(newEntry) yes and done. setValue(params) no as the newEntry calculation in the if case depends on setValue(params) being done.